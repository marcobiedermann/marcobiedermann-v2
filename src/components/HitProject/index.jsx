import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';

const HitProject = props => {
  const { slug, title } = props;

  return (
    <div>
      <figure>
        <figcaption>
          <h3>
            <Link to={`/projects/${slug['en-US']}`}>{title['en-US']}</Link>
          </h3>
        </figcaption>
      </figure>
    </div>
  );
};

HitProject.propTypes = {
  slug: PropTypes.shape({
    'en-US': PropTypes.string,
  }).isRequired,
  title: PropTypes.shape({
    'en-US': PropTypes.string,
  }).isRequired,
};

export default HitProject;
