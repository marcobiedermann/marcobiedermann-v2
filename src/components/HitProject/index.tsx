import { Link } from 'gatsby';
import React from 'react';

export interface HitProjectProps {
  slug: {
    'en-US': string;
  };
  title: {
    'en-US': string;
  };
}

const HitProject: React.FC<HitProjectProps> = (props) => {
  const { slug, title } = props;

  return (
    <div>
      <figure>
        <figcaption>
          <h3>
            <Link to={`/projects/${slug['en-US']}`}>{title['en-US']}</Link>
          </h3>
        </figcaption>
      </figure>
    </div>
  );
};

export default HitProject;
