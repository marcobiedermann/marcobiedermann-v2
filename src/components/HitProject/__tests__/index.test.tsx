/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import HitProject from '..';

describe('HitProject component', () => {
  it('should render correctly', () => {
    const project = {
      slug: {
        'en-US': 'demo-project',
      },
      title: {
        'en-US': 'Demo Project',
      },
    };
    const tree = renderer.create(<HitProject {...project} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
