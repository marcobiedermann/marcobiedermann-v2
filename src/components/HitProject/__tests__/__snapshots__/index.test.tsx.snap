// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HitProject component should render correctly 1`] = `
<div>
  <figure>
    <figcaption>
      <h3>
        <a
          href="/projects/demo-project"
        >
          Demo Project
        </a>
      </h3>
    </figcaption>
  </figure>
</div>
`;
